tests/unit/test_network_connections.py

```diff
@@ -317,13 +317,19 @@ def do_connections_validate_ifcfg(self, input_connections, **kwargs):
             content_current = kwargs.get("initscripts_content_current", None)
             if content_current:
                 content_current = content_current[idx]
-            c = IfcfgUtil.ifcfg_create(
-                connections, idx, content_current=content_current
+            warnings = []
+            config = IfcfgUtil.ifcfg_create(
+                connections,
+                idx,
+                content_current=content_current,
+                warn_fcn=lambda msg: warnings.append(msg),
             )
             # pprint("con[%s] = \"%s\"" % (idx, connections[idx]['name']), c)
-            exp = kwargs.get("initscripts_dict_expected", None)
-            if exp is not None:
-                self.assertEqual(exp[idx], c)
+            expected_config = kwargs.get("initscripts_dict_expected", None)
+            if expected_config is not None:
+                self.assertEqual(expected_config[idx], config)
+            expected_warnings = kwargs.get("initscripts_expected_warnings", [])
+            self.assertEqual(expected_warnings, warnings)
```

Comment:
This commit is taken from https://github.com/linux-system-roles/network/commit/ac7cc6fe94cc4fd856a323b6d0fe26e3309c2c6b which is submitted by @tyll . I honestly do not know how to  change the author into @tyll .  @tyll , please change the author's name if you'd like to.

Reply:
I don't know how you applied the patch. If you use git to do it, it keeps the author. If you added my repo and fetched from it (which is my recommendation), you should be able to run `git cherry-pick  f6d6a79a01b8004527e15ef84715571aaa85f78a` to get the commit. If you do not want to do this, you can download and apply the patch:

```
wget https://github.com/linux-system-roles/network/commit/ac7cc6fe94cc4fd856a323b6d0fe26e3309c2c6b.patch
git am ac7cc6fe94cc4fd856a323b6d0fe26e3309c2c6b.patch
```

To change the author of the latest commit (you might need to rebase interactively to edit the commit if it is not the latest), use

```
git commit --amend --author "Till Maas ..."
```

Regarding the commit message, you also removed my Signed-off-by line. And about
"The unit test is missing warnings check when testing with initscritps provider, add support for that."  

- The summary already states "Unit tests/initscripts: Support checking warnings", this text does not seem to add any additional information, does it?
- "The unit test" is wrong, since the patch is about the TestValidator class, not one specific unit test. The unit test framework could be considered correct.
- "is missing warnings check" - checks would be part of a specific test, support for checks (which is added here) part of the framework. So "is missing support to check for warnings" might be correct
	- If this is about adjusting an existing unit test, it would be ok to write "The unit test for route devices is missing a check for warnings" but this is not the case here.
- If you want to add additional context to the commit message, I guess this would be the fact that the unit test to be added will need this to be written, so something like "This will be needed to test that warnings are generated when specifying routes in connections without an interface name." I am not convinced, that this is adding much value to the commit message, though.

Reply:
I just knew that adding the suffix `.patch` to the commit will get the raw patch. Thanks for sharing. I just felt weird for me to change the author without the permission from the author. For example, I created a commit which had all the code changes from another author, then I felt uncomfortable to sign off on behalf of another author and left the author's name as another author (without the explicit permission from another author).

Indeed, the commit message that I added did not introduce any new information.